fleet_vehicle_inspection/models/fleet_vehicle_inspection.py

```diff
@@ -0,0 +1,196 @@
+    )
+
+    note = fields.Html("Notes", states=READONLY_STATES)
+
+    inspection_line_ids = fields.One2many(
+        "fleet.vehicle.inspection.line",
+        "inspection_id",
+        string="Inspection Lines",
+        copy=True,
+        auto_join=True,
+        states=READONLY_STATES,
+    )
+
+    result = fields.Selection(
+        [("todo", "Todo"), ("success", "Success"), ("failure", "Failure")],
+        "Inspection Result",
+        default="todo",
+        compute="_compute_inspection_result",
+        readonly=True,
+        copy=False,
+        store=True,
+    )
+
+    @api.depends("inspection_line_ids", "state")
+    def _compute_inspection_result(self):
+        for inspection in self:
+            if inspection.inspection_line_ids:
+                if inspection.inspection_line_ids.filtered(
+                    lambda x: x.result == "todo"
+                ):
+                    inspection.result = "todo"
+                    continue
+                elif inspection.inspection_line_ids.filtered(
+                    lambda x: x.result == "failure"
+                ):
+                    inspection.result = "failure"
+                else:
+                    inspection.result = "success"
+            else:
+                inspection.result = "todo"
+
+    @api.model
+    def create(self, vals):
+        if vals.get("name", "New") == "New":
+            if vals.get("direction") == "out":
+                vals["name"] = (
+                    self.env["ir.sequence"].next_by_code("fleet.vehicle.inspection.out")
+                    or "/"
+                )
+            else:
+                vals["name"] = (
+                    self.env["ir.sequence"].next_by_code("fleet.vehicle.inspection.in")
+                    or "/"
+                )
+        return super(FleetVehicleInspection, self).create(vals)
+
+    @api.multi
+    def button_cancel(self):
+        for inspection in self:
+            if inspection.state not in ["draft", "confirmed"]:
+                continue
+            inspection.write({"state": "cancel"})
+        return True
+
+    @api.multi
+    def button_confirm(self):
+        for inspection in self:
+            if inspection.inspection_line_ids:
+                if inspection.state not in ["draft", "cancel"]:
+                    continue
+                if inspection.inspection_line_ids.filtered(
+                    lambda x: x.result == "todo"
+                ):
+                    raise UserError(
+                        _(
+                            "Inspection cannot be completed. "
+                            "There are uninspected items."
+                        )
+                    )
+                inspection.write({"state": "confirmed"})
+            else:
+                raise UserError(
+                    _("Inspection cannot be completed. There are no inspected items.")
+                )
+        return True
```

Comment:
Following up on the previous comment, I'd do something like this here

```suggestion
    def button_confirm(self):
        if (
            any(not rec.inspection_line_ids for rec in self)
            or any(line.result == "todo" for line in self.mapped("inspection_line_ids"))
        ):
            raise UserError(_(
                "Inspection cannot be completed. "
                "There are uninspected items."
            ))
        if any(rec.state not in ["draft", "cancel"] for rec in self):
            raise ValidationError(_(
                "Only inspections in 'draft' or 'cancel' states can be confirmed"
            ))
        return self.write({"state": "confirmed"})            
```